glslc/test/working_directory.py

```diff
@@ -0,0 +1,166 @@
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirEqNoArg(expect.ErrorMessage):
+    """Tests -working-directory=<empty>. Behavior cribbed from Clang."""
+
+    glslc_args = ['-working-directory=']
+    expected_error = ['glslc: error: no input files\n']
+
+
+EMPTY_SHADER_IN_SUBDIR = Directory(
+    'subdir', [File('shader.vert', 'void main() {}')])
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirEqNoArgCompileFile(expect.ValidNamedObjectFile):
+    """Tests -working-directory=<empty> when compiling input file."""
+
+    environment = Directory('.', [EMPTY_SHADER_IN_SUBDIR])
+    glslc_args = ['-c', '-working-directory=', 'subdir/shader.vert']
+    # output file should be generated into subdir/.
+    expected_object_filenames = ('subdir/shader.vert.spv',)
+
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirEqNoArgLinkFile(expect.ValidNamedObjectFile):
+    """Tests -working-directory=<empty> when linking input file."""
+
+    environment = Directory('.', [EMPTY_SHADER_IN_SUBDIR])
+    glslc_args = ['-working-directory=', 'subdir/shader.vert']
+    # output file should be generated into the current directory, not
+    # subdir/, where the source is.
+    expected_object_filenames = ('a.spv',)
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirCompileFile(expect.ValidNamedObjectFile):
+    """Tests -working-directory=<dir> when compiling input file."""
+
+    environment = Directory('.', [EMPTY_SHADER_IN_SUBDIR])
+    glslc_args = ['-c', '-working-directory=subdir', 'shader.vert']
+    # output file should be generated into subdir/.
+    expected_object_filenames = ('subdir/shader.vert.spv',)
+
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirLinkFile(expect.ValidNamedObjectFile):
+    """Tests -working-directory=<dir> when linking input file."""
+
+    environment = Directory('.', [EMPTY_SHADER_IN_SUBDIR])
+    glslc_args = ['-working-directory=subdir', 'shader.vert']
+    # output file should be generated into the current directory, not
+    # subdir/, where the source is.
+    expected_object_filenames = ('a.spv',)
+
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirCompileFileOutput(expect.ValidNamedObjectFile):
+    """Tests -working-directory=<dir> when compiling input file and specifying
+    output filename."""
+
+    environment = Directory('.', [
+        Directory('subdir', [
+            Directory('bin', []),
+            File('shader.vert', 'void main() {}')
+        ])
+    ])
+    glslc_args = ['-c', '-o', 'bin/spv', '-working-directory=subdir',
+                  'shader.vert']
+    # output file should be generated into subdir/bin/.
+    expected_object_filenames = ('subdir/bin/spv',)
+
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirLinkFileOutput(expect.ValidNamedObjectFile):
+    """Tests -working-directory=<dir> when linking input file and specifying
+    output filename."""
+
+    environment = Directory('.', [
+        EMPTY_SHADER_IN_SUBDIR,
+        Directory('bin', [])
+    ])
```

Comment:
Then -working-directory doesn't affect the linked output file at all.  All those linked-file tests are basically just testing that -working-directory has no impact on linked-file designation.  After you check this in, I think I'd like to group them all together and name them better -- they only differ by their -wd and -o arguments.